software/deploy/deploy.go

```diff
@@ -289,3 +293,91 @@ func getAirflowUILink(deploymentID string, deploymentURLs []houston.DeploymentUR
 	}
 	return ""
 }
+
+func isDagOnlyDeploymentEnabled(appConfig *houston.AppConfig) bool {
+	return appConfig != nil && appConfig.Flags.DagOnlyDeployment
+}
+
+func isDagOnlyDeploymentEnabledForDeployment(deploymentInfo *houston.Deployment) bool {
+	return deploymentInfo != nil && deploymentInfo.DagDeployment.Type == houston.DagOnlyDeploymentType
+}
+
+func validateIfDagDeployURLCanBeConstructed(deploymentInfo *houston.Deployment) error {
+	_, err := config.GetCurrentContext()
+	if err != nil {
+		return fmt.Errorf("could not get current context! Error: %w", err)
+	}
+	if deploymentInfo == nil || deploymentInfo.ReleaseName == "" {
+		return errInvalidDeploymentID
+	}
+	return nil
+}
+
+func getDagDeployURL(deploymentInfo *houston.Deployment) string {
+	c, _ := config.GetCurrentContext()
+	return fmt.Sprintf("deployments.%s/%s/upload", c.Domain, deploymentInfo.ReleaseName)
+}
+
+func DagsOnlyDeploy(houstonClient houston.ClientInterface, appConfig *houston.AppConfig, deploymentID, dagsParentPath string, dagDeployURL *string, cleanUpFiles bool) error {
+	// Throw error if the feature is disabled at Houston level
+	if !isDagOnlyDeploymentEnabled(appConfig) {
+		return ErrDagOnlyDeployDisabledInConfig
+	}
+
+	// Throw error if the feature is disabled at Deployment level
+	deploymentInfo, err := houston.Call(houstonClient.GetDeployment)(deploymentID)
+	if err != nil {
+		return fmt.Errorf("failed to get deployment info: %w", err)
+	}
+	if !isDagOnlyDeploymentEnabledForDeployment(deploymentInfo) {
+		return errDagOnlyDeployNotEnabledForDeployment
+	}
+
+	uploadURL := ""
+	if dagDeployURL == nil {
+		// Throw error if the upload URL can't be constructed
+		err = validateIfDagDeployURLCanBeConstructed(deploymentInfo)
+		if err != nil {
+			return err
+		}
+		uploadURL = getDagDeployURL(deploymentInfo)
+	} else {
+		uploadURL = *dagDeployURL
+	}
+
+	dagsPath := filepath.Join(dagsParentPath, "dags")
+	dagFiles := fileutil.GetFilesWithSpecificExtension(dagsPath, ".py")
+
+	// Alert the user if dags folder is empty
+	if len(dagFiles) == 0 && config.CFG.ShowWarnings.GetBool() {
+		i, _ := input.Confirm("Warning: No DAGs found. This will delete any existing DAGs. Are you sure you want to deploy?")
+		if !i {
+			return ErrEmptyDagFolderUserCancelledOperation
+		}
+	}
+
+	start := time.Now()
+
+	// Generate the dags tar
+	err = fileutil.Tar(dagsPath, dagsParentPath)
+	if err != nil {
+		return err
+	}
+	if cleanUpFiles {
+		defer os.Remove(dagsParentPath + "/dags.tar")
+	}
+
+	// Gzip the tar
+	err = fileutil.GzipFile(dagsParentPath+"/dags.tar", dagsParentPath+"/dags.tar.gz")
+	if err != nil {
+		return err
```

Comment:
Again, the good way to get an error from this function call was to provide an invalid path. But then, the flow would have never reached here due to line 362. Can we leave this line without test?
Or please let me know if there is a better way to test this.
An alternate would be to pass destination parameters for both Tar() and GzipFile() as different arguments for DagsOnlyDeploy function, but that would be unnecessary and it will be just to cover this line.

Reply:
You can do something like this for GzipFile and Tar functions - https://github.com/astronomer/astro-cli/blob/main/cloud/deploy/deploy.go#L65

you can then mock the functions to return happy and error paths

Reply:
I assumed that this is a less preferable way, as it involves non-local variables. But if we are fine with this way, let me add the tests